contrib/devtools/github-merge.py

```diff
@@ -69,6 +70,27 @@ def ask_prompt(text):
     print("",file=stderr)
     return reply
 
+def tree_sha512sum():
+    files = sorted(subprocess.check_output([GIT, 'ls-tree', '--full-tree', '-r', '--name-only', 'HEAD']).splitlines())
+    overall = hashlib.sha512()
+    for f in files:
+        intern = hashlib.sha512()
+        fi = open(f, 'rb')
+        while True:
+            piece = fi.read(65536)
+            if piece:
+                intern.update(piece)
+            else:
+                break
+        fi.close()
```

Comment:
Nit: use `with` instead of `open`/`close`:
```Python
with open(f, 'rb') as fi:
    while True:
        piece = fi.read(65536)
        if piece:
            intern.update(piece)
        else:
            break
```